webapi/src/main/scala/org/knora/webapi/messages/v2/responder/resourcemessages/ResourceMessagesV2.scala

```diff
@@ -84,31 +84,30 @@ case class ReadValueV2(valueIri: IRI, valueContent: ValueContentV2) extends IOVa
       * Converts this value to JSON-LD.
       *
       * @param targetSchema the target schema.
-      * @param settings the application settings.
+      * @param settings     the application settings.
       * @return a JSON-LD representation of this value.
       */
     def toJsonLD(targetSchema: ApiV2Schema, settings: SettingsImpl): JsonLDValue = {
         val valueContentAsJsonLD = valueContent.toJsonLDValue(targetSchema, settings)
 
-        // Is the value represented as a JSON-LD object in the target schema?
-        valueContentAsJsonLD match {
-            case jsonLDObject: JsonLDObject =>
-                // Yes. Is it just an IRI?
-                if (JsonLDUtil.isIriValue(jsonLDObject)) {
-                    // Yes. Leave it as is.
-                    jsonLDObject
-                } else {
-                    // No. Add the value's IRI and type.
-                    JsonLDObject(
-                        jsonLDObject.value +
-                            ("@id" -> JsonLDString(valueIri)) +
-                            ("@type" -> JsonLDString(valueContent.valueType.toString))
-                    )
+        // In the complex schema, add the value's IRI and type to the JSON-LD object that represents it.
```

Comment:
@SepidehAlassi This fixes the error you were getting, sorry about that.

Reply:
no problem, thank you for quick fix!